.github/workflows/5_pr_check.yml

```diff
@@ -0,0 +1,542 @@
+       fi
+
+    - name: Check Wazuh manager start
+      run: | 
+        for NODE in "${{ env.MANAGER_NODES }}"; do
+          for i in {1..20}; do
+            TOKEN=$(curl -s -u ${{ env.API_USERNAME }}:${{ env.API_PASSWORD }} -k -X GET "https://127.0.0.1:55000/security/user/authenticate?raw=true")
+            services="`curl -k -s -X GET "https://127.0.0.1:55000/cluster/$NODE/status?pretty=true" -H  "Authorization: Bearer ${TOKEN}" | jq -r .data.affected_items | grep running | wc -l`"
+            if [[ $services -gt 9 ]]; then
+              echo "Wazuh Manager $NODE Services: ${services}"
+              echo "OK"
+            else
+              curl -k -X GET "https://127.0.0.1:55000/cluster/$NODE/status?pretty=true" -H  "Authorization: Bearer ${TOKEN}" | jq -r .data.affected_items
+              echo "Wazuh Manager $NODE Services: ${services}. Retrying in 30s"
+              [ $i -lt 20 ] && sleep 30
+            fi
+          done
+        done
+        
+    - name: Check Wazuh dashboard service URL
+      run: |
+        for i in {1..20}; do
+          echo "Checking Wazuh dashboard (Attempt $i/20)"
+          STATUS=$(curl -k -s -o /dev/null -w "%{http_code}" -u ${{ env.INDEXER_USERNAME }}:${{ env.INDEXER_PASSWORD }} "https://127.0.0.1:443/app/status")
+          echo "Current status: $STATUS"
+          if [[ "$STATUS" == "200" ]]; then
+            echo "Wazuh dashboard is UP"
+            exit 0
+          elif [[ "$STATUS" == "429" || "$STATUS" == "503" ]]; then
+            echo "Dashboard is busy or initializing (Status $STATUS). Retrying in 30s"
+          else
+            echo "Unexpected status $STATUS. Retrying in 30s"
+          fi
+          sleep 30
+        done
+        echo "Error: Dashboard did not reach 200 status in time."
+        exit 1
+
+    - name: Modify Docker endpoint into Wazuh agent docker-compose.yml file
+      run: sed -i "s/<WAZUH_MANAGER_IP>/$(ip addr show docker0 | grep 'inet ' | awk '{print $2}' | cut -d'/' -f1)/g" wazuh-agent/docker-compose.yml
+
+    - name: Edit Wazuh agent docker-compose file
+      shell: bash
+      env:
+        WAZUH_REGISTRY: ${{ env.WAZUH_REGISTRY }}
+      run: |
+        TARGET_FILE="wazuh-agent/docker-compose.yml"
+        if [ -f "$TARGET_FILE" ]; then
+          echo "Updating registry in $TARGET_FILE to: ${{ env.WAZUH_REGISTRY }}"
+          sed -i "s|wazuh/wazuh-|${{ env.WAZUH_REGISTRY }}/wazuh/wazuh-|g" "$TARGET_FILE"
+        else
+          echo "File $TARGET_FILE not found"
+          exit 1
+        fi
+
+    - name: Start Wazuh agent
+      run: docker compose up -d
+      working-directory: ./wazuh-agent
+
+    - name: Check Wazuh agent enrollment
+      run: |
+        sleep 20
+        curl -k -s -X GET "https://127.0.0.1:55000/agents?pretty=true" -H  "Authorization: Bearer ${{env.TOKEN}}"
+      env:
+        TOKEN: $(curl -s -u ${{ env.API_USERNAME }}:${{ env.API_PASSWORD }} -k -X GET "https://127.0.0.1:55000/security/user/authenticate?raw=true")
```

Comment:
Add a validation for the number of agents and replace the sleep with a retry every 10 seconds (5 retries).
Consider this change for the single-node test.